ports/raspberrypi/common-hal/picodvi/Framebuffer.c

```diff
@@ -143,8 +143,12 @@ void common_hal_picodvi_framebuffer_construct(picodvi_framebuffer_obj_t *self,
 
     bool color_framebuffer = color_depth >= 8;
     const struct dvi_timing *timing = NULL;
-    if ((width == 640 && height == 480) ||
-        (width == 320 && height == 240)) {
+    if ((width == 640 && height == 240)) {
+        timing = &dvi_timing_640x480p_60hz;
```

Comment:
640x240 -> 640x480? This does not preserve the aspect ratio. If that's what you want you could add this to the lines 640x480 and 320x240 `if` arm.

Reply:
That's right, this mode will double pixels vertically but not horizontally. Lots of "classic" computers have modes with extremely non-square pixels such as CGA's 640x200 mode.

Reply:
so you'd rather the resolution changing code looked more like this?
```c
     if ((width == 640 && height == 480) ||
-        (width == 320 && height == 240)) {
+        (width == 320 && height == 240) ||
+        (width == 640 && height == 240)
+        ) {
         timing = &dvi_timing_640x480p_60hz;
     } else if ((width == 800 && height == 480) ||
-               (width == 400 && height == 240)) {
+               (width == 400 && height == 240) ||
+               (width == 800 && height == 240) ||
+               ) {
         timing = &dvi_timing_800x480p_60hz;
     } else {
```
so that there's only one "case" for each timing mode? I can do that.

Reply:
Yup, that's what I meant, since all those cases choose the same timing value.